tests/test_core.py

```diff
@@ -1278,14 +1277,14 @@ def test_exceptions_allowed(self):
     print('empty_size: %d' % empty_size)
     print('fake_size: %d' % fake_size)
     print('disabled_size: %d' % disabled_size)
-    self.assertEqual(empty_size, fake_size)
+    # empty list acts the same a full disable
+    self.assertEqual(empty_size, disabled_size)
     # big change when we disable exception catching of the function
     self.assertGreater(size - empty_size, 0.01 * size)
     # full disable can remove a little bit more
-    self.assertLess(disabled_size, empty_size)
+    self.assertLess(disabled_size, fake_size)
```

Comment:
It still seems to contain some of the previous PR?

Reply:
These changes are actually part of this one now.

Reply:
If empty list act as a full disable and `empty_size` is equal to `fake_size` as you wrote, what is `disabled_size` here? It also says this is full disable... Which one is real full disable? `empty_size` of `disabled_size`? Does setting `EXCEPTION_CATCHING_ALLOWED` to a nonempty list have any other effect?

Reply:
disabled_size is when no options are given and exceptions are therefore completely disabled.

With this new change settings `EXCEPTION_CATCHING_ALLOWED` to empty list is the same as completely disabled.

Settings `EXCEPTION_CATCHING_ALLOWED` to a non-empty list internally sets `DISABLE_EXCEPTION_CATCHING=0` so all the JS-side assumes exceptions can be caught.  The only difference is in the underlying call to clang which either limits (or does not limit) the functions which can catch (IIUC).